frontend/src/components/history/MatchList.tsx

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from 'react'
+import { useAppSelector } from '../../store/types'
+import { UserData } from '../../types/UserData'
+import { MatchData } from '../../types/MatchData'
+import styles from './MatchList.module.css'
+
+const MatchList = () => {
+    const [matchHistory, setMatchHistory] = useState<MatchData[]>([])
+    const userData = useAppSelector((state) => state.user.userData) as UserData
+
+    useEffect(() => {
+        const fetchMatches = async () => {
+            try {
+                const res = await fetch(
+                    `http://localhost:8080/api/match/user/`,
+                    { credentials: 'include' }
+                )
+                const resjson = await res.json()
+                setMatchHistory(resjson)
+                console.log(resjson)
+            } catch (error) {
+                console.log(error)
+            }
+        }
+        fetchMatches()
+    }, )
```

Comment:
```suggestion
    }, []) 
```
To prevent continuous fetching

Reply:
Fixed. Could you explain what that means someday? I don't understand well how this all works yet